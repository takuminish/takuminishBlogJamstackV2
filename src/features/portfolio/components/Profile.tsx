import Box from "@mui/material/Box";
import { Profile as ProfileType } from "../../../pages";
import Typography from "@mui/material/Typography";
import IconButton from "@mui/material/IconButton";
import React from "react";
import GitHubIcon from "@mui/icons-material/GitHub";

type Props = {
  profile: ProfileType;
};

function Profile(props: Props) {
  const { profile } = props;

  return (
    <Box
      sx={{
        textAlign: "center",
      }}
    >
      <img src={profile.profileImage.url} width="150" height="150" />
      <Typography variant="h4" sx={{ fontWeight: 700, textAlign: "center" }}>
        {profile.name}
      </Typography>
      <Typography variant="body1">
        <div
          dangerouslySetInnerHTML={{
            __html: profile.introduction,
          }}
        ></div>
      </Typography>
      <IconButton href={profile.githubUrl}>
        <GitHubIcon />
      </IconButton>
    </Box>
  );
}

export default Profile;
