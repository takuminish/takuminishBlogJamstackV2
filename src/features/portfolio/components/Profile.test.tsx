import React from "react";
import { render } from "@testing-library/react";
import Profile, { Props } from "./Profile";

const createDOM = () => {
  const props: Props = {
    profile: {
      fieldId: "profile",
      name: "name1",
      introduction: "introduction1",
      githubUrl: "https://testfwaefa.com",
      qiitaUrl: "https://testfwaefa.com",
      profileImage: {
        url: "https://testfwaefa/comofile.jpeg",
        height: 400,
        width: 400,
      },
    },
  };
  return render(<Profile profile={props.profile} />);
};

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});
