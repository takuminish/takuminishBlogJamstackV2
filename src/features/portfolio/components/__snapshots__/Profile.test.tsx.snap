// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-xi606m"
  >
    <img
      height="150"
      src="https://testfwaefa/comofile.jpeg"
      width="150"
    />
    <h4
      class="MuiTypography-root MuiTypography-h4 css-uqrixu-MuiTypography-root"
    >
      name1
    </h4>
    <div>
      introduction1
    </div>
    <a
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      href="https://testfwaefa.com"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="GitHubIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
